python/cudf/cudf/core/_compat.py

```diff
@@ -4,11 +4,7 @@
 from packaging import version
 
 PANDAS_VERSION = version.parse(pd.__version__)
-PANDAS_EQ_200 = PANDAS_VERSION == version.parse("2.0.0")
```

Comment:
I'm a bit confused. What are our criteria for keeping or removing these? I assumed that only `PANDAS_GE_200` would be safe to remove, because that's the only thing we can assume by migrating to pandas 2.x. Do we not care about introducing new test failures with pandas 2.0.0 (for example) that were previously xfail'd for that version?

Reply:
From reading the code, it seems like there could be an assumption that we only support the latest patch release from a given 2.0.*, 2.1.*, etc? We need to write out the rules for whatever we're expecting in the developer docs (or possibly the public docs on pandas compatibility).

Reply:
After discussing offline, we landed on a new method of testing as described in dev docs of this PR.